src/main/java/SYSC4806/Model/Program.java

```diff
@@ -17,6 +14,8 @@
     @GeneratedValue(strategy= GenerationType.AUTO)
     private long id;
     private String name;
+
+    @OneToMany
```

Comment:
I believe this should be a @ManyToMany as a course can belong to many programs (also in Course, you specified this relationship as such). Following the same tutorial, I think this needs a @mappedBy attribute.